docs/core/knowledge-bases.rst

```diff
@@ -0,0 +1,416 @@
+:desc: Leverage information from knowledge bases inside conversations using ActionQueryKnowledgeBase
+       in open source bot framework Rasa.
+
+.. _knowledge_bases:
+
+Knowledge Base Actions
+======================
+
+.. edit-link::
+
+.. warning::
+   This feature is experimental.
+   We introduce experimental features to get feedback on implementations, so do please try it out.
+   However, the functionality might be changed or removed in the future.
+   If you have feedback (positive or negative) please comment in the `community forum <https://forum.rasa.com>`_.
+
+.. contents::
+   :local:
+
+
+Knowledge base actions enable you to handle the following kind of dialogues:
+
+.. image:: ../_static/images/knowledge-base-example.png
+
+As you can see users might want to obtain detailed information about certain objects, such as restaurants or hotels,
+during a conversation.
+They want to obtain details, such as, if an restaurant has outside seating or how expensive the restaurant is.
+In order to answer those user requests domain knowledge is needed.
+Hard-coding the information would not help as the information are subject to change.
+Additionally, users do not only refer to objects by their names, but also use terms, such as "the first one" or "it",
```

Comment:
I think this is the key point that the KB helps with so should be mentioned first.

Reply:
not really the KB, but more the smart usage of slots